pyiode/objects/equation.pyx

```diff
@@ -666,9 +690,10 @@ cdef class Equation:
         sample = self.sample
         tests = self.tests
         indent = " " * len("Equation(")
+        ident_lec = " " * len("lec = ")
 
         s = [f"endogenous = {self.endogenous}"]
-        s += [f"lec = {self.lec}"]
+        s += ["lec = {}".format(self.lec.replace('\n', '\n' + indent + ident_lec))]
```

Comment:
1. Why do use `format()` instead of f-string ? I see no reason to prefer format() to f-string.
2. Why can't you use `_format_lec_string()` here ? Why not adding a extra argument "extra_indent" to `_format_lec_string()` with default to '' ?

Reply:
Something like `s += [f"lec = {self.lec.replace('\n', '\n' + indent + ident_lec)}"]` would have yielded a *"backslashes not allowed in f-strings"* error (cf. similar examples [here](https://stackoverflow.com/questions/44780357/how-can-i-use-newline-n-in-an-f-string-to-format-output)). Using `.format()` was a workaround to that. 

One can also first introduce an extra variable such as `newline_fix = '\n' + indent + ident_lec` or even substitute `\n` by `chr(10)` and then use f-strings.

Reply:
`str.format()` was kept for backward compatibility but should not be used in any new Python code running on Python >= 3.6
The less ugly solution here is to proceed in 2 steps:
```python
    lec = self.lec.replace('\n', '\n' + indent + ident_lec)
    s += [f"lec = {lec}"]
```

But maybe another solution is to add an extra argument `extra_indent` to your function `_format_lec_string` and then call:
```python
    ident_lec = " " * len("lec = ")
    (...)
    s += [f"lec = {_format_lec_string(self.lec, ident_lec)}"]
```
?

Reply:
I tried the extra argument suggestion in `_format_lec_string(self.lec, extra_indent=indent_eq+indent_lec))` but this behaves rather *unstable*. 

I finally figured out its strange behaviour: the first `self.lec` argument implicitly already calls `_format_lec_string` a first time, so in total, the function `_format_lec_string` gets called two times, which complicates getting the indentation right. 

The following does behave as expected, though:

```python
def __str__(self) -> str:
    sample = self.sample
    tests = self.tests
    indent_eq = " " * len("Equation(")
    indent_lec = " " * len("lec = ")

    s = [f"endogenous = {self.endogenous}"]
    s += [f"lec = {self._format_lec_string(self.c_equation.get_lec().decode(), extra_indent=indent_eq+indent_lec)}"]
```

The most clean approach still seems to be:
```python
lec = self.lec.replace('\n', '\n' + indent_eq + ident_lec)
s += [f"lec = {lec}"]
```